source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
I'm not surprised by the interesting convergence behavior. I do find these cases where something doesn't happen "as one would expect" often make for good teaching situations because one can explain what is going wrong and why, and then how to fix it. That's what I did in the section in the manual on the other (thermochemical) van Keken benchmark. I typically find it more instructive to also show the situations that don't work, before showing how to make it work.

I have no objections to #1865 or a variation thereof.